concent_api/core/views.py

```diff
@@ -1241,6 +1378,239 @@ def store_message_and_message_status(
         receive_message_status.save()
 
 
+def store_subtask(
+    task_id:                        str,
+    subtask_id:                     str,
+    provider_public_key:            bytes,
+    requestor_public_key:           bytes,
+    state:                          Enum,
+    next_deadline:                  int,
+    task_to_compute:                message.TaskToCompute                = None,
+    report_computed_task:           message.ReportComputedTask           = None,
+    ack_report_computed_task:       message.AckReportComputedTask        = None,
+    reject_report_computed_task:    message.RejectReportComputedTask     = None,
+    subtask_results_accepted:       message.tasks.SubtaskResultsAccepted = None,
+    subtask_results_rejected:       message.tasks.SubtaskResultsRejected = None,
+):
+    assert isinstance(task_id,              str)
+    assert isinstance(subtask_id,           str)
+    assert isinstance(provider_public_key,  bytes)
+    assert isinstance(requestor_public_key, bytes)
+    assert state in Subtask.SubtaskState
+
+    if state in Subtask.SUBTASK_ACTIVE_STATES:
+        assert next_deadline is not None
+    elif state in Subtask.SUBTASK_PASSIVE_STATES:
+        assert next_deadline is None
+
+    provider  = Client.objects.get_or_create_full_clean(provider_public_key)
+    requestor = Client.objects.get_or_create_full_clean(requestor_public_key)
+
+    subtask = Subtask(
+        task_id         = task_id,
+        subtask_id      = subtask_id,
+        provider        = provider,
+        requestor       = requestor,
+        state           = state.name,
+        next_deadline   = datetime.datetime.fromtimestamp(next_deadline, timezone.utc),
+    )
+
+    add_messages_to_subtask(
+        subtask,
+        task_to_compute,
+        report_computed_task,
+        ack_report_computed_task,
+        reject_report_computed_task,
+        subtask_results_accepted,
+        subtask_results_rejected,
+    )
+
+    subtask.full_clean()
+    subtask.save()
+
+
+def update_subtask(
```

Comment:
It should be possible to create single function which will create or update subtask directly, no like currently created one which is using either one or another.